sematic/api_client.py

```diff
@@ -706,15 +708,17 @@ def _put(
     return response.json()
 
 
-def validate_server_compatibility(use_cached: bool = True) -> None:
+def validate_server_compatibility(
+    use_cached: bool = True,
+) -> Dict[Literal["server", "min_client_supported"], List[int]]:
     """Check that the client is compatible with the server.
 
     Raises an error if the server and client are incompatible, or if this can't be
     verified.
     """
-    global _validated_client_version
-    if _validated_client_version and use_cached:
-        return
+    global _server_version_metadata
+    if _server_version_metadata and use_cached:
```

Comment:
Would the server ever respond with an empty payload?
```suggestion
    if _server_version_metadata is not None and use_cached:
```

Reply:
no, but the very first time this is called, `_server_version_metadata` will be `None`